src/components/common/dashboard/OperationComponent.vue

```diff
@@ -0,0 +1,386 @@
+                (operation.value.missingApprovals.users.includes(props.enrollmentID) ||
+                    operation.value.missingApprovals.groups.includes(props.role)) &&
+                !isRejected;
+
+            const sentApprove = ref(store.getters.treatedOperations.approved.includes(operation.value.operationId));
+            const sentReject = ref(store.getters.treatedOperations.rejected.includes(operation.value.operationId));
+            const provideReason = ref(false);
+            const selectedReason = ref("");
+            const writtenReason = ref("");
+            const isMyOperation = operation.value.initiator == props.enrollmentId;
+            const showWatchOption = !isMyOperation && isPending;
+
+            const isAdmin = props.role == Role.ADMIN;
+            const username = ref("...");
+
+            const isWatched = ref(props.watched);
+
+            const dateFormatOptions = {
+                weekday: "short",
+                year: "numeric",
+                month: "numeric",
+                day: "numeric",
+                hour: "2-digit",
+                minute: "2-digit",
+            };
+
+            const params = ref([] as string[]);
+            const title = ref("");
+            const shownOpId = operation.value.operationId.substring(0, 4);
+
+            const initiatedTimestamp = new Date(operation.value.initiatedTimestamp).toLocaleString("en-US", dateFormatOptions);
+            const lastUpdateTimestamp = new Date(operation.value.lastModifiedTimestamp).toLocaleString("en-US", dateFormatOptions);
+
+            const showDetails = ref(false);
+
+            onBeforeMount(async () => {
+                if (isAdmin) {
+                    getNameByEnrollmentId();
+                }
+                createDisplayObjects();
+            });
+
+            const type = computed(() => {
+                switch (operation.value.transactionInfo.contractName) {
+                    case UC4Identifier.CONTRACT_MATRICULATION:
+                        return "Matriculation";
+                    case UC4Identifier.CONTRACT_ADMISSION:
+                        return "Course Admission";
+                    default:
+                        return "";
+                }
+            });
+
+            const statusColor = computed(() => {
+                switch ((props.operation as Operation).state) {
+                    case OperationStatus.PENDING:
+                        return "bg-blue-300";
+                    case OperationStatus.FINISHED:
+                        return "bg-green-300";
+                    case OperationStatus.REJECTED:
+                        return "bg-red-300";
+                }
+            });
+
+            watch(selectedReason, () => {
+                writtenReason.value = selectedReason.value;
+                if (writtenReason.value == RejectionReasons.OTHER) {
+                    writtenReason.value = "";
+                }
+            });
```

Comment:
writtenReason seems like its always user defined, but is it not.. should be renamed or refactored

Reply:
changed to `finalReason` in https://github.com/upb-uc4/ui-web/pull/785/commits/3817f74d9dfdaaba0941ad70bd2e9baaf2eef347